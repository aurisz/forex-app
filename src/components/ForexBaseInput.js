import React from 'react';
import PropTypes from 'prop-types';
import { Row, Col, Input } from 'reactstrap';

const ForexBaseInput = ({ value, onChange }) => (
  <div className="header">
    <Row>
      <Col>
        <p>USD - United States Dollars</p>
      </Col>
    </Row>
    <Row>
      <Col>
        <h4>USD</h4>
      </Col>
      <Col>
        <Input
          type="number"
          name="baseValue"
          value={value}
          onChange={onChange}
          placeholder="Enter Currency Value Here"
        />
      </Col>
    </Row>
  </div>
);

ForexBaseInput.propTypes = {
  value: PropTypes.number.isRequired,
  onChange: PropTypes.func.isRequired
};

export default ForexBaseInput;
