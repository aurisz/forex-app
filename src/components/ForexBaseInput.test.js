import React from 'react';
import { shallow, mount } from 'enzyme';
import renderer from 'react-test-renderer';

import ForexBaseInput from './ForexBaseInput';

describe('<ForexBaseInput />', () => {
  it('renders without crashing', () => {
    const wrapper = shallow(
      <ForexBaseInput value={10} onInputChange={jest.fn()} />
    );
    expect(wrapper.exists()).toBe(true);
  });

  it('should accept value of integer', () => {
    const tree = renderer
      .create(<ForexBaseInput value={20} onInputChange={jest.fn()} />)
      .toJSON();
    expect(tree).toMatchSnapshot();
  });

  it('should accept value of float', () => {
    const tree = renderer
      .create(<ForexBaseInput value={10.5} onInputChange={jest.fn()} />)
      .toJSON();
    expect(tree).toMatchSnapshot();
  });

  it('should accept value of empty strings', () => {
    const tree = renderer
      .create(<ForexBaseInput value="" onInputChange={jest.fn()} />)
      .toJSON();
    expect(tree).toMatchSnapshot();
  });

  it('Simulates onInputChange event', () => {
    const wrapper = mount(
      <ForexBaseInput value={10} onInputChange={jest.fn()} />
    );
    const input = wrapper.find('input#baseValueInput');

    input.simulate('change', {
      target: { value: 10 }
    });

    expect(input.props().value).toEqual(10);

    wrapper.unmount();
  });
});
