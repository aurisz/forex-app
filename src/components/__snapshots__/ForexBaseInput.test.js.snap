// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<ForexBaseInput /> should accept value of empty strings 1`] = `
<div
  className="header"
>
  <div
    className="row"
  >
    <div
      className="col"
    >
      <p>
        <em>
          USD - United States Dollars
        </em>
      </p>
    </div>
  </div>
  <div
    className="row"
  >
    <div
      className="col"
    >
      <h3>
        USD
      </h3>
    </div>
    <div
      className="col"
    >
      <input
        className="form-control"
        id="baseValueInput"
        name="baseValue"
        onChange={[MockFunction]}
        placeholder="Enter Base Currency Value Here"
        step="any"
        type="number"
        value=""
      />
    </div>
  </div>
</div>
`;

exports[`<ForexBaseInput /> should accept value of float 1`] = `
<div
  className="header"
>
  <div
    className="row"
  >
    <div
      className="col"
    >
      <p>
        <em>
          USD - United States Dollars
        </em>
      </p>
    </div>
  </div>
  <div
    className="row"
  >
    <div
      className="col"
    >
      <h3>
        USD
      </h3>
    </div>
    <div
      className="col"
    >
      <input
        className="form-control"
        id="baseValueInput"
        name="baseValue"
        onChange={[MockFunction]}
        placeholder="Enter Base Currency Value Here"
        step="any"
        type="number"
        value={10.5}
      />
    </div>
  </div>
</div>
`;

exports[`<ForexBaseInput /> should accept value of integer 1`] = `
<div
  className="header"
>
  <div
    className="row"
  >
    <div
      className="col"
    >
      <p>
        <em>
          USD - United States Dollars
        </em>
      </p>
    </div>
  </div>
  <div
    className="row"
  >
    <div
      className="col"
    >
      <h3>
        USD
      </h3>
    </div>
    <div
      className="col"
    >
      <input
        className="form-control"
        id="baseValueInput"
        name="baseValue"
        onChange={[MockFunction]}
        placeholder="Enter Base Currency Value Here"
        step="any"
        type="number"
        value={20}
      />
    </div>
  </div>
</div>
`;
